### PR TITLE
[doc] Add generalized issue priority definitions

### PR DESCRIPTION
**Pull request by moidx**
This commit addresses the RFC referenced by #17104.

Fixes https://github.com/lowRISC/opentitan/issues/17104

### CODE REVIEW

File: doc/project/priority_definitions.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+The following definitions are used:
+
+*  **Work streams**: OpenTitan sub-projects which typically have their own associated working group, e.g.: discrete chip, integrated IP.
+*  **Milestones**: Milestones (M1, M2 etc)  associated with a specific work stream.
+*  **Core function**: Core function of OpenTitan in any of the relevant engineering domains (HW, SW, security).
+
+### Priority Definitions Table
+
+
+<table>
```
Comment (mundaym) [inline doc/project/priority_definitions.md:17]: HTML tables don't render well in mdbook so we might need to convert this. @pamaury for visibility.

**Review by mundaym — APPROVED** [review]
Thanks @moidx.

File: doc/project/priority_definitions.md (hunk 1 above)
Comment (pamaury) [reply doc/project/priority_definitions.md:17]: It's probably better to using Github Markdown tables (they are supported by mdbook). See
[https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-tables](the doc)

Comment (moidx) [reply doc/project/priority_definitions.md:17]: Is there a way to render markdown inside a table cell? That is the only reason I added html. I will try to render something with GitHub markdown.

Comment (moidx) [reply doc/project/priority_definitions.md]: I switched to Markdown format. I had to add some partial rows to make it work. Once the new rendering is in place we can revisit this to make it better.

**Pull request merged**